src/test/java/cart/IntegrationTest.java

```diff
@@ -0,0 +1,114 @@
+package cart;
+
+import static io.restassured.RestAssured.given;
+
+import cart.service.ProductService;
+import cart.service.dto.ProductRequest;
+import io.restassured.RestAssured;
+import org.apache.catalina.connector.Response;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.jdbc.Sql;
+
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+class IntegrationTest {
+
+    @Autowired
+    private ProductService productService;
+
+    @LocalServerPort
+    private int port;
+
+    @BeforeEach
+    void setUp() {
+        RestAssured.port = port;
+    }
+
+    @Test
+    @DisplayName("관리자 도구 페이지 접근 테스트")
+    void productList() {
+        given()
+                .when()
+                .get("/admin")
+                .then()
+                .statusCode(200);
+    }
+
+    @Test
+    @DisplayName("상품 추가 테스트")
+    void createProduct() {
+        final ProductRequest productRequest = new ProductRequest("TEST",
+                "https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png", 4000);
+
+        given()
+                .body(productRequest).log().all()
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .when()
+                .post("/product")
+                .then().log().all()
+                .statusCode(201);
+    }
+
+    @Test
+    @DisplayName("상품 추가 테스트 - 값이 비어있는 경우")
+    void createProductValueEmpty() {
+        final String jsonStrNameEmpty = "{ \"price\":\"321321\", \"imageUrl\":\"apple.png\"}";
+
+        given()
+                .body(jsonStrNameEmpty)
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .when()
+                .post("/product")
+                .then()
+                .statusCode(Response.SC_BAD_REQUEST);
```

Comment:
```suggestion
        given()
                .body(jsonStrNameEmpty)
                .contentType(MediaType.APPLICATION_JSON_VALUE)
                .when()
                .post("/product")
                .then()
                .statusCode(HttpStatus.BAD_REQUEST.value());
```

이거 httpStatusCode를 비교하는 부분인데 `org.apache.catalina.connector.Response` 패키지의 `Response.SC_BAD_REQUEST`를 사용해주신 이유가 있나요?

Reply:
아 해당 코드를 작성할 당시에 BAD_REQUEST가 enum으로 지정된 패키지를 잘 찾지 못해서,

`org.apache.catalina.connector.Response.SC_BAD_REQUEST`를 사용했었어요.

HttpStatus.BAD_REQUEST.value()가 더 일반적인 표현이 되겠네요.